src/SimpleCircuitBreaker.php

```diff
@@ -30,24 +30,21 @@ public function __construct(
     public function call($service, callable $fallback)
     {
         $transaction = $this->initTransaction($service);
-
         try {
             if ($this->isOpened()) {
-                if ($this->canAccessService($transaction)) {
-                    $this->moveStateTo(States::HALF_OPEN_STATE, $service);
+                if (!$this->canAccessService($transaction)) {
```

Comment:
can you explain to me what this change update on the current workflow?

Reply:
When HALF_OPEN is detected, we directly try to recontact the service (using the half open state settings since we switched the state).
Previously we had OPEN call detecting the HALF_OPEN state before returning, and we had to perform an additional call for HALF_OPEN state to actually occur. This resulted in the possible states after a `call` execution, and a delay because service was not called at the moment it should have.

Now as soon as the threshold has expired, we switch to HALF_OPEN and request the service right away, besides at the end of a call only two states are possible OPEN or CLOSED.

Reply:
all right, make sense!

This is something I'll backport asap on resiliency!

Finally, it's a good idea to have 2 libraries :rofl: